crates/primitives/src/block.rs

```diff
@@ -125,74 +103,17 @@ impl Block {
         Some(BlockWithSenders { block: self, senders })
     }
 
-    /// Returns whether or not the block contains any blob transactions.
-    #[inline]
-    pub fn has_blob_transactions(&self) -> bool {
-        self.body.iter().any(|tx| tx.is_eip4844())
-    }
-
-    /// Returns whether or not the block contains any EIP-7702 transactions.
-    #[inline]
-    pub fn has_eip7702_transactions(&self) -> bool {
-        self.body.iter().any(|tx| tx.is_eip7702())
-    }
-
-    /// Returns an iterator over all blob transactions of the block
-    #[inline]
-    pub fn blob_transactions_iter(&self) -> impl Iterator<Item = &TransactionSigned> + '_ {
-        self.body.iter().filter(|tx| tx.is_eip4844())
-    }
-
-    /// Returns only the blob transactions, if any, from the block body.
-    #[inline]
-    pub fn blob_transactions(&self) -> Vec<&TransactionSigned> {
-        self.blob_transactions_iter().collect()
-    }
-
-    /// Returns an iterator over all blob versioned hashes from the block body.
-    #[inline]
```

Comment:
none of those should be removed I think

Reply:
I moved them to `BlockBody`, but I'll keep them and call the `BlockBody`'s methods.